include/alpaka/atomic/AtomicHierarchy.hpp

```diff
@@ -0,0 +1,125 @@
+/**
+* \file
+* Copyright 2014-2015 Benjamin Worpitz
+*
+* This file is part of alpaka.
+*
+* alpaka is free software: you can redistribute it and/or modify
+* it under the terms of the GNU Lesser General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+*
+* alpaka is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU Lesser General Public License for more details.
+*
+* You should have received a copy of the GNU Lesser General Public License
+* along with alpaka.
+* If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include <alpaka/core/Positioning.hpp>
+#include <alpaka/atomic/Traits.hpp>                 // AtomicOp
+
+#include <type_traits>                              // std::conditional, std::is_same
+
+namespace alpaka
+{
+    namespace atomic
+    {
+        namespace atomicHierarchy
+        {
+            class Empty0{};
+            class Empty1{};
+            class Empty2{};
+        }
+        //#############################################################################
+        //! build a single class to inherit from different atomic implementations
+        //
+        //  \tparam TGridAtomic  atomic implementation for atomics in a single grid
+        //  \tparam TBlockAtomic atomic implementation for atomics in a single block
+        //  \tparam TThreadAtomic atomic implementation for atomics in a single thread
```

Comment:
@BenjaminW3 Yes you are right your description is better, I misunderstood your comment. Know I am with you^^